packages/wouter/src/use-browser-location.js

```diff
@@ -30,7 +30,12 @@ export const useLocationProperty = (fn, ssrFn) =>
   useSyncExternalStore(subscribeToLocationUpdates, fn, ssrFn);
 
 const currentSearch = () => location.search;
-export const useSearch = () => useLocationProperty(currentSearch);
+
+export const useSearch = ({ ssrSearch } = {}) =>
+  useLocationProperty(
+    currentSearch,
+    ssrSearch ? () => ssrSearch : currentSearch
+  );
```

Comment:
I've been thinking, what are the cases when we would want to use `currentSearch` on a server? Maybe it's better to have default value of `""` for `ssrSearch`?

If a developer for some reason forgets to set `ssrSearch` while rendering, they will receive an error now, while they might instead render the app with empty params and then receive a hydration error on the client.